subprojects/code-quality/src/main/groovy/org/gradle/api/plugins/quality/PmdExtension.java

```diff
@@ -50,7 +50,7 @@ public PmdExtension(Project project) {
     }
```

Comment:
:o: The default version is now >6.0.0. Maybe we should enable incremental analysis at line 48. (previously  it was set to false by @big-guy)

Reply:
`getIncrementalAnalysis()` is marked as Incubating
https://github.com/gradle/gradle/blob/199f9673fc5cac48a2e469e74b13f5a8ec575a6c/subprojects/code-quality/src/main/groovy/org/gradle/api/plugins/quality/PmdExtension.java#L225-L226

Should it be left as is?

Reply:
IIRC, there were problems with earlier versions of the incremental analysis. I'm not sure if it's 100% reliable, but if it is, we could do this as a separate PR.

I would tie incremental analysis to the tool version requested, so it would be a slightly harder change than just flipping false to true.  That would look something like...
```
this.incrementalAnalysis = project.getObjects().property(Boolean.class).convention(project.provider(() -> VersionNumber.parse(getToolVersion()).compareTo(VersionNumber.parse("6.0.0")) >= 0);
```

WRT Incubating, it was introduced in 5.6 and we usually leave things as incubating for a few releases before marking it stable.  In this case, it's probably stable enough, but let's do that separately from the upgrade.